src/typecheck/mod.rs

```diff
@@ -146,8 +149,8 @@ pub fn env_add_term(
     let RichTerm { term, pos } = rt;
 
     match term.as_ref() {
-        Term::Record(bindings, _) | Term::RecRecord(bindings, ..) => {
-            for (id, t) in bindings {
+        Term::Record(record) | Term::RecRecord(record, ..) => {
+            for (id, t) in &record.fields {
```

Comment:
Why do we need `&` now?

Reply:
I was also initially a little surprised this was required, but essentially: accessing a field of a borrowed struct implicitly deferences that struct in order to find the field, so if `x: &X` then `x.y` actually results in `(*x).y`.

That is to say that if we have:

```rust
struct MyStruct {
    x: String
}

let my_struct = MyStruct { x: "hello".into() };
let borrowed = &my_struct;
let val = borrowed.x;
```

Then `val` would have type `String` - i.e. it's not borrowed. This example would pas type checking, but would fail the borrow checker as it attempts to move the value of `x` out of a borrowed struct. If we instead wrote `let val = &borrowed.x;` then we'd get a reference & the borrow checker would be happy.

Previously we had `&HashMap<_, _>`, so no explicit borrow was required, but now that we're accessing a field of a struct, we need to make it explicit that we're not trying to move that field.

Reply:
Thank you for the explanation!